huaweicloud/services/er/resource_huaweicloud_er_flow_log.go

```diff
@@ -0,0 +1,412 @@
+		updateFlowLogPath := client.ResourceBaseURL() + updateFlowLogHttpUrl
+		updateFlowLogPath = strings.ReplaceAll(updateFlowLogPath, "{er_id}", d.Get("instance_id").(string))
+		updateFlowLogPath = strings.ReplaceAll(updateFlowLogPath, "{flow_log_id}", d.Id())
+
+		updateFlowLogOpts := golangsdk.RequestOpts{
+			KeepResponseBody: true,
+			OkCodes: []int{
+				200,
+			},
+		}
+		updateFlowLogOpts.JSONBody = utils.RemoveNil(buildUpdateFlowLogBodyParams(d))
+		_, err = client.Request("PUT", updateFlowLogPath, &updateFlowLogOpts)
+		if err != nil {
+			return diag.Errorf("error updating ER flow log: %s", err)
+		}
+
+		err = flowLogWaitingForStateCompleted(ctx, d, meta, d.Timeout(schema.TimeoutUpdate))
+		if err != nil {
+			return diag.Errorf("error waiting for the update of flow log (%s) to complete: %s", d.Id(), err)
+		}
+	}
+
+	if d.HasChanges("enabled") {
+		enabled := d.Get("enabled").(bool)
+		if enabled {
+			err = updateFlowLogState(client, d, "enable")
+		} else {
+			err = updateFlowLogState(client, d, "disable")
+		}
+
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		err = flowLogWaitingForStateCompleted(ctx, d, meta, d.Timeout(schema.TimeoutUpdate))
+		if err != nil {
+			return diag.Errorf("error waiting for the update of flow log (%s) to complete: %s", d.Id(), err)
+		}
+	}
+
+	return resourceFlowLogRead(ctx, d, meta)
+}
+
+func resourceFlowLogDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	client, err := cfg.ErV3Client(region)
+	if err != nil {
+		return diag.Errorf("error creating ER v3 Client: %s", err)
+	}
+
+	deleteFlowLogHttpUrl := "enterprise-router/{er_id}/flow-logs/{flow_log_id}"
+	deleteFlowLogPath := client.ResourceBaseURL() + deleteFlowLogHttpUrl
+	deleteFlowLogPath = strings.ReplaceAll(deleteFlowLogPath, "{er_id}", d.Get("instance_id").(string))
+	deleteFlowLogPath = strings.ReplaceAll(deleteFlowLogPath, "{flow_log_id}", d.Id())
+
+	deleteFlowLogOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			202,
+		},
+	}
+	_, err = client.Request("DELETE", deleteFlowLogPath, &deleteFlowLogOpt)
+	if err != nil {
+		return diag.Errorf("error deleting flow log: %s", err)
+	}
+
+	stateConf := &resource.StateChangeConf{
+		Pending:      []string{"PENDING"},
+		Target:       []string{"COMPLETED"},
+		Refresh:      flowLogStatusRefreshFunc(d, meta, nil),
+		Timeout:      d.Timeout(schema.TimeoutDelete),
+		Delay:        10 * time.Second,
+		PollInterval: 5 * time.Second,
+	}
+	_, err = stateConf.WaitForStateContext(ctx)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	return nil
+}
+
+func flowLogStatusRefreshFunc(d *schema.ResourceData, meta interface{}, targets []string) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		resp, err := getFlowLogInfo(d, meta)
+		if err != nil {
+			if _, ok := err.(golangsdk.ErrDefault404); ok && len(targets) < 1 {
```

Comment:
if `targets` is only used here, I think we can define it as bool type

Reply:
fixed